conf/app.ini

```diff
@@ -46,6 +46,10 @@ DOMAIN = localhost
 ROOT_URL = %(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
 HTTP_ADDR =
 HTTP_PORT = 3000
+; Local (DMZ) URL for gogs workers (such as ssh update) accessing web service.
+; In most cases you may leave it empty. Defaults: http://localhost:HTTP_PORT/
+; You may need to set it only if ssh server node is not the same as http node.
+LOCAL_ROOT_URL =
```

Comment:
Okay, I need some clarification what's the purpose of this template? `LOCAL_ROOT_URL = http://localhost:%(HTTP_PORT)s/` does it land into `custom/conf/app.ini`? Who and what actually parses `http://localhost:%(HTTP_PORT)s/` and replaces `%(HTTP_PORT)s` with the value. Finally doesn't the `MustString("....")` serve the same purpose?